bom/pom.xml

```diff
@@ -536,6 +545,18 @@
             </plugin>
          </plugins>
       </pluginManagement>
+      <plugins>
+         <plugin>
+            <groupId>org.sonatype.plugins</groupId>
+            <artifactId>nexus-staging-maven-plugin</artifactId>
+            <extensions>true</extensions>
+            <configuration>
+               <nexusUrl>${jboss.releases.nexus.url}</nexusUrl>
+               <serverId>${jboss.releases.repo.id}</serverId>
+               <autoReleaseAfterClose>true</autoReleaseAfterClose>
```

Comment:
Well yes and no...

If you import this BOM, you won't trigger this plugin on your project (note it's in `/build/plugins`). The problem is when you use the BOM as your parent, than yes - the you will run the staging plugin when you invoke `mvn deploy`. But please note that today we have the same problem - if you use this BOM as a parent of your project you will use JBoss Repositories from `/distributionManagement` since they are also inherited. 

So yes, it's wrong but our current approach is also wrong.

Now diving into more technical things - this plugin needs to hook into `default-deploy` execution. This is done by `<extensions>true</extensions>` in the configuration. Now if we don't like this approach, we will need to disable the default Deploy plugin (adding an execution with goal `none`) and invoke deployment separately using command line. Something like this: `mvn install && mvn nexus-staging-maven-plugin:deploy -D...<all required parameters to get this thing work>`. This probably needs some experiments since I've never done it before.

As I wrote in my email on internal mailing list - I believe we should use Nexus Staging and automate the release process as a whole. That's a much better approach than automatically closing the staging repository here.